nengo/params.py

```diff
@@ -416,6 +426,47 @@ def validate(self, instance, function_info):
         super(FunctionParam, self).validate(instance, function)
 
 
+class CopyableObject(object):
```

Comment:
> I'm not sure if this should be a mixin class.

It encapsulates the copy related code from other `NengoObject` instead of mixing it all in one class (and same functions). To me that seems a bit clearer, but I think I'm ok with it either way as long as we don't have any other classes with parameters that should support copies.